frontend/src/v5/ui/routes/dashboard/projects/federations/federationsList/federationsList.component.tsx

```diff
@@ -126,14 +114,14 @@ export const FederationsList = ({
 				</DashboardListHeader>
 				<DashboardList>
 					{!isEmpty(sortedList) ? (
-						sortedList.map((federation, index) => (
+						sortedList.map((federation, index) => (federation.hasStatsPending ? (
+							<SkeletonListItem delay={index / 10} key={federation._id} />
+						) : (
```

Comment:
I'm not against this, but why the change?

Reply:
Also why moving **onFavouriteChange** ?

Reply:
1. This way you don't have to pass in 'index' as a prop to FederationListItem. If the index were to change (i.e. by adding or deleting a previous item) then the component would rerender. Also I'm not a fan of conditionally rendering outside of the main return (if that makes sense) because you have to make sure all your hooks occur before this otherwise you get vague errors that always take me a minute to decipher.

2. I could just wrap it in a useCallback() and pass it in as a prop but it seems overcomplex when you can just define the function lower in the DOM where it is actually used